src/Mod/Sketcher/Gui/EditModeCoinManagerParameters.h

```diff
@@ -461,7 +571,8 @@ struct CoinMapping
 
     //* These map a MF index (second index) within a coin layer (first index) for points or curves
     // to a GeoId */
-    std::vector<std::vector<int>> CurvIdToGeoId;   // conversion of SoLineSet index to GeoId
+    std::vector<std::vector<std::vector<int>>>

```

Comment:
It is not a change for this PR, but this is getting way too complex. In that point we probably should introduce some proper structures for the layers with proper names, methods etc which would be easier to follow later. 

API could look something like that (just a concept):
```
layer(0).type(SubLayer::External).curveToGeo(curveIndex)
```

Now we have to deal with 3-level deep structures which are not easy to read and remember, stuff like `swcs[l * 4 + t]`.